### PR TITLE
Parsing optimizations

### PR DESCRIPTION
**Pull request by 38**

### CODE REVIEW

File: src/utils/general/ParseTools.cpp [hunk 1]
```diff
@@ -162,25 +82,46 @@ bool isHeaderLine(const string &line) {
...
-	if (memcmp(tmp.c_str(), "browser", 7) == 0) {
-		return true;
-	}
-	if (memcmp(tmp.c_str(), "track", 5) == 0) {
-		return true;
-	}
-	if (memcmp(tmp.c_str(), "visibility", 10) == 0) {
-		return true;
+	if(line.length() > 4) {
+		uint32_t peek = *(uint32_t*)line.c_str() | 0x20202020u;
```
Comment (jmarshall) [inline src/utils/general/ParseTools.cpp:86]: This makes unportable assumptions about endianness and alignment.

File: src/utils/FileRecordTools/Records/Bed3Interval.cpp [hunk 2]
```diff
@@ -32,26 +32,18 @@ bool Bed3Interval::initFromFile(SingleLineDelimTextFileReader *fileReader)
...
-void Bed3Interval::print(string &outBuf) const
-{
-	ostringstream s;
-	s << _chrName;
-	s << "\t";
-	s << _startPos;
-	s << "\t";
-	s << _endPos;
-	outBuf.append(s.str());
+inline void print_reocord(const string& name, CHRPOS start, CHRPOS end, string& buf) {
```
Comment (arq5x) [inline src/utils/FileRecordTools/Records/Bed3Interval.cpp:35]: Can you change to `print_record`?

**Review by arq5x — COMMENTED** [review]
Hi @38 - can you apply the new `print_record` functionality to all of the *Interval classes?

File: src/utils/general/ParseTools.cpp (hunk 1 above)
Comment (38) [reply src/utils/general/ParseTools.cpp:86]: I am not worry about this. I think bedtools is already unportable, so in this situation, I think it's reasonable for trading portability at this point for performance. In fact, we are not expecting this runs on cpu other than X86. In the future, we should refactor this completely.

Comment (jmarshall) [reply src/utils/general/ParseTools.cpp:88]: Which code already in bedtools do you think is unportable?

Comment (arq5x) [reply src/utils/general/ParseTools.cpp:99]: Thanks for the comment, @jmarshall - @38 made that block dependent on the detection of X86 CPU.

**Comment by arq5x** [comment]
Thank you, @38!

**Pull request merged**